### DISCUSSION TITLE
Recording from meeting with Hadley

### DISCUSSION THREAD

**Discussion by yjunechoe** [ggplot2-extenders/ggplot-extension-club#52] [General]
Hi all - the recording from our meeting with Hadley today (thanks Hadley for the permission!) is now live on YouTube: https://youtu.be/kjjcgdkowXs

I'm also hoping to use this thread to continue some of our discussions from today and to give those of us who couldn't make it a space to share their thoughts. Please leave a comment here if you'd like to follow up with anything! :)

**Comment by yjunechoe** [comment]
As for myself:

1) The full quote from the [Extending ggplot2 vignette](https://ggplot2.tidyverse.org/articles/extending-ggplot2.html) that I was referring to in my question is:

    > Next we write a layer function. Unfortunately, due to an **early design mistake** I called these either stat_() or geom_(). A better decision would have been to call them layer_() functions: that’s a more accurate description because every layer involves a stat and a geom.

2) On the topic of non-ggplot2-like implementation of the Grammar of Graphics, I'm reminded of [AlgebraOfGraphics.jl](https://aog.makie.org/stable/) from Julia. They were *very* intentional about not trying to simply replicate the ggplot2 interface in Julia, and have done some creative re-imaging of the syntax and the internals. There's also an early effort in the Clojure data science community to make a grammar of graphics library ([ggclj](https://github.com/kirahowe/ggclj)) - I actually got a chance to meet the Clojure devs on this a while ago and they were imagining a very data-oriented interface, where the internal plot representation (think `str(ggplot())`) *is* the code (which the user can specify up to various degrees of verbosity).

**Comment by EvaMaeRey** [comment]
> Hi all - the recording from our meeting with Hadley today (thanks Hadley for the permission!) is now live on YouTube: https://youtu.be/kjjcgdkowXs

Thanks for uploading and sharing here @yjunechoe!

**Comment by EvaMaeRey** [comment]
Just looking at sentiments in transcript - tidytext making a quick look so easy. 🙃  'Plot' is classified as negative as in to scheme I guess - which shows up a bunch of course - and is obviously mis-classification in this context... 

data.frame(text = text) |>
  tidytext::unnest_tokens(output = word, input = text) |>
  anti_join(tidytext::stop_words) |>
  inner_join(tidytext::get_sentiments("bing"), 
             by = "word", 
             relationship = "many-to-many") %>% 
  count(word, sentiment) ->
hadley_meetup_bing_sentiments 
hadley_meetup_bing_sentiments

![image](https://github.com/user-attachments/assets/65166c1d-1b89-4632-a3f0-0bc71eaaf504)

hadley_meetup_bing_sentiments |>
  ggplot() + 
  aes(id = word, fill = sentiment, area = n) + 
  ggcirclepack::geom_circlepack() + 
  ggcirclepack::geom_circlepack_text() + 
  facet_wrap(~sentiment)

![image](https://github.com/user-attachments/assets/43ca79e3-846d-45df-83d8-f8573fb3b96b)

last_plot() + 
  aes(id = sentiment, 
      label = after_stat(area)) + 
  aes(size = 1)

![image](https://github.com/user-attachments/assets/f8ade208-a758-47d4-b09c-de07de27c53e)

**Reply by teunbrand** [reply]
"plot" has negative sentiment ☹️
plots spark joy!

**Reply by EvaMaeRey** [reply]
Overwrite to positive sentiment!  Just another 'look at your data' moment.  'bump' is another interesting negative classification.  😯

**Reply by teunbrand** [reply]
I guess they're just usually in different context. Outside in the wilds, 'plot' usually refers to a storyline and 'plotting' is scheming.
Also bumps are inconvenient to drive over with your car or unattrictive to have on your skin.

**Reply by EvaMaeRey** [reply]
Slightly higher resolution. 

![image](https://github.com/user-attachments/assets/50596b2c-5516-48e9-bbe3-f92181d50590)

**Reply by EvaMaeRey** [reply]
Maybe 'bing' classifications isn't the best choice. 

@lepennec's ggwordcloud also a great exploratory tool.  First time trying out - seems incredibly straight forward! 

![image](https://github.com/user-attachments/assets/e7b10588-d100-4389-8282-7bdba463c547)
